wlauto/common/android/device.py

```diff
@@ -670,6 +670,18 @@ def get_device_model(self):
         except KeyError:
             return None
 
+    def is_network_connected(self, network_host='www.google.com'):
+        self.logger.debug('Checking for internet connectivity.')
+        output = self.execute('ping -q -w 1 -c 1 {}'.format(network_host),
```

Comment:
The ping for busybox requires a rooted device whereas the built in does not. The targets we have tested so far all have a ping command built in and I would expect this to be included as standard for android devices. If this is still not acceptable than let me know.